src/ReflectionMethod.php

```diff
@@ -47,6 +43,10 @@ public function isStatic()
 
     public function getDocComment()
     {
-        return preg_replace('/@return Generated\\\\/', '@return ', $this->method->getDocComment());
+        return preg_replace(
+            '/@(return|param) Generated\\\\/',
+            '@${1} ',
+            $this->method->getDocComment()
+        );
```

Comment:
What about union returns/vars?

 - `@return Collection|Generated\SomeInterface[]`
 - `@param Collection|Generated\SomeInterface[]`
 - `@var Collection|Generated\SomeInterface[]`

This is due to the restriction of both the Collection framework and untyped arrays.